aiida/manage/configuration/__init__.py

```diff
@@ -59,6 +59,24 @@
 BACKEND_UUID = None  # This will be set to the UUID of the profile as soon as its corresponding backend is loaded
 
 
+def check_rabbitmq_version():
+    """Check the version of RabbitMQ that is being connected to and emit warning if the version is not compatible.
+
+    Versions 3.8 and above are not compatible with AiiDA with default configuration.
+    """
+    from packaging.version import parse
+
+    from aiida.cmdline.utils import echo
+    from aiida.manage.manager import get_manager
+
+    url = 'https://github.com/aiidateam/aiida-core/wiki/RabbitMQ-version-to-use'
+    communicator = get_manager().get_communicator()
+    version = parse(communicator.server_properties['version'].decode('utf-8'))
+    if version >= parse('3.8'):
+        echo.echo_warning(f'Connected to RabbitMQ v{version} which is known to cause problems)')
```

Comment:
I would be even more explicit, e.g. add "Please don't use AiIDA with this version of RabbitMQ!!! Unexepected problems might occur like crashes of long-running workflows, or multiple submissions of the same job." or something that strongly discourages further use of AiiDA until this is fixed.